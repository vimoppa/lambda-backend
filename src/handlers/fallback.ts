import { successResponse } from '../libs/response';
import { Response } from '../libs/response/types';
import wrap from '../libs/wrap';

const handler = async (): Promise<Response> => {
  const paths = [
    // {
    //   path: "/v1/store/create",
    //   method: "POST",
    //   example: "/v1/store/create",
    // },
  ] as { path: string; method: string; example: string }[];

  return successResponse({
    message: "This endpoint doesn't exist, check `paths` for a list of all endpoints.",
    paths,
  });
};

export default wrap(handler);
