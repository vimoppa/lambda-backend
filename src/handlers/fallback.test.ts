import fallback from './fallback';

describe('snapshot of error provided', () => {
  it('executes as expected', async () => {
    const response = await fallback({
      headers: {},
    } as any);
    expect(response).toMatchSnapshot();
  });
});
