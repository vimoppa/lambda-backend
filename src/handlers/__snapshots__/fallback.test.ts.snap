// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot of error provided executes as expected 1`] = `
{
  "body": "{"message":"This endpoint doesn't exist, check \`paths\` for a list of all endpoints.","paths":[]}",
  "headers": {
    "Access-Control-Allow-Credentials": "true",
    "Access-Control-Allow-Origin": "*",
    "Content-Type": "application/json",
  },
  "statusCode": 200,
}
`;
